bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -1031,8 +989,11 @@ public void deleteSnapshots(String regex) throws IOException {
    */
   @Override
   public void deleteSnapshots(Pattern pattern) throws IOException {
-    for (SnapshotDescription snapshotDescription : listSnapshots(pattern)) {
-      deleteSnapshot(snapshotDescription.getName());
+    if (pattern != null && !pattern.matcher("").matches()) {
+      for (SnapshotDescription description : listSnapshots(pattern)) {
+        tableAdminClientWrapper.deleteBackupAsync(
+            getClusterName().getClusterId(), description.getName());
```

Comment:
deleteSnapshots is synchronous, so it shouldnt return until it finished all of the deletes. So I think you need await all of the futures.

for (SnapshotDescription description : listSnapshots(pattern)) {
  futures.add(tableAdminClientWrapper.deleteBackupAsync());
}
Futures.allAsList(futures).get();